### DISCUSSION TITLE
Is it possible to make id_token_hint optional on front channel logout request?

### DISCUSSION THREAD

**Discussion by arahlf** [keycloak/keycloak#17149] [Q&A]
I'm trying to integrate Keycloak behind Azure B2C (using an OIDC client) and while the login works great, the logout from B2C doesn't actually fully log out of Keycloak. When B2C makes the front channel logout request to Keycloak, it doesn't pass the id_token_hint and is technically getting that logout confirmation page but the user doesn't see that since it's hidden in an iframe. It doesn't seem that B2C supports adding the id_token_hint to federated single sign out requests either.

Is it possible to make passing the id_token_hint to Keycloak optional somehow?

**Comment by LMonert** [comment]
I think that it is not possible. They describes the changes here https://www.keycloak.org/2022/04/keycloak-1800-released.html in detail and it seems that setting the "--spi-login-protocol-openid-connect-legacy-logout-redirect-uri=true" backwards compatibility option only enables the use of the old redirect_uri parameter but the id token stays as a requirement. 
I have had the same problem with a different application and had to change the setup so that users now click the logout button on the new confirmation screen.  I don't know anything about Azure B2C but maybe you can use backchannel logout?

**Reply by sschu** [reply]
If I am not mistaken starting with KC19 you can omit `id_token_hint` and even suppress the confirmation screen by passing `--spi-login-protocol-openid-connect-legacy-logout-redirect-uri=true` and `--spi-login-protocol-suppress-logout-confirmation-screen=true`.

**Reply by arahlf** [reply]
Thanks @sschu that works!

**Reply by lsimone** [reply]
[not any more](https://www.keycloak.org/docs/latest/upgrading/index.html#support-for-legacy-redirect_uri-parameter-and-spi-options-has-been-removed) (v26.3.3)
